journalpump/senders/elasticsearch_opensearch_sender.py

```diff
@@ -72,6 +73,8 @@ class _EsOsLogSenderBase(LogSender):
 
     _DEFAULT_MAX_SENDER_INTERVAL = 10.0
 
+    _INDICIES_URL_REDACTION_REGEXP = r"(\w*?://[A-Za-z0-9\-_]*)(:)([A-Za-z0-9\-_]*)(@)"
```

Comment:
The user and password may be percent-encoded or have other punctuation in them, from reading https://datatracker.ietf.org/doc/html/rfc3986#section-3.2.1 I suggest this regex which matches characters from the "unreserved", "pct-encoded", and "sub-delims" elements in the username and password parts.

```suggestion
    _INDICIES_URL_REDACTION_REGEXP = r"(\w*?://[A-Za-z0-9\-._~%!$&'()*+,;=]*)(:)([A-Za-z0-9\-._~%!$&'()*+,;=]*)(@)"
```